### PR TITLE
feat: add leadbox component

### LINKED ISSUE #155: Component: Leadbox
### Design
https://www.figma.com/file/FWlI63N28LP7BxqNqj1bGw/sy-004-(Design!)?node-id=3675%3A19858

### Things to consider
- Should the content be "hardcoded" (yet configurable) or a simple slot that can display anything?

### Expected outcome
A new component called `LeadBox` that complies to the design and is fully responsive.

### PR DESCRIPTION
**Pull request by hofmann-felix**
URL: https://satellytescommain21751-featleadbox.gtsb.io/

**What's included?**
- added leadbox component
- added corresponding text to each leadbox 

Note for the reviewers: Please also have a look at the English translations of the leadbox text

Ticket: https://github.com/satellytes/satellytes.com/issues/155

### CODE REVIEW

File: src/locales/de/translations.json
Comment (georgiee) [inline src/locales/de/translations.json]: Hm, I don't want to receive applications. We have the following text included already.

> Or send your application to career@satellytes.com

I think we should replace my email with that and remove the text before it.
Screen Shot 2021-11-08 at 09 44 23

```suggestion
    "leadbox": {
      "title": "Deine Stelle nicht dabei?",
      "subtitle": "Initiativbewerbung",
      "text": "Georgios Kaleadis, HR Manager",
      "mail": "career@satellytes.com"
```

File: src/locales/de/translations.json
Comment (georgiee) [inline src/locales/de/translations.json:80]: 👍

File: src/locales/de/translations.json
Comment (georgiee) [inline src/locales/de/translations.json]: 👍 @abdolg are you good with the public email here for inquiries ?

File: src/locales/de/translations.json
Comment (georgiee) [inline src/locales/de/translations.json]: 👍

File: src/locales/en/translations.json
Comment (georgiee) [inline src/locales/en/translations.json]:
```suggestion
      "mail": "career@satellytes.com"
```

Comment (georgiee) [reply src/locales/en/translations.json]: as said at the german translation, we have a dedicated email for applications.

Comment (georgiee) [inline src/locales/en/translations.json]: text is good 👍

File: src/locales/en/translations.json
Comment (georgiee) [inline src/locales/en/translations.json:80]: 👍

File: src/locales/en/translations.json
Comment (georgiee) [inline src/locales/en/translations.json]: 👍

File: src/locales/en/translations.json
Comment (georgiee) [inline src/locales/en/translations.json]: 👍

File: src/pages/clients.tsx
Comment (georgiee) [inline src/pages/clients.tsx:37]: that's slick. very easy to understand how to craft and pass in a leadbox 👍

File: src/templates/blog-post.tsx
Comment (georgiee) [inline src/templates/blog-post.tsx]: wrong key, I will suggest the fixed key (`blogpost`) and suggest according changes to the wrong key (being `blogpos`)
```suggestion
  const leadbox = {
    title: t('blogpost.leadbox.title'),
    link: t('blogpost.leadbox.link'),
    linkTo: '/career/',
    icon: <Astronaut />,
  };

```

In the future I think it would be awesome to pass in some unique text per blog post to make the leadbox aware of the surrounding content. Like "have you learned something about gatsby, then join us to learn more" or so.

File: src/locales/en/translations.json
Comment (georgiee) [inline src/locales/en/translations.json]:
```suggestion
  "blogpost": {
```

File: src/locales/de/translations.json
Comment (georgiee) [inline src/locales/de/translations.json]:
```suggestion
  "blogpost": {
```

Comment (georgiee) [reply src/locales/de/translations.json]: typo (I also suggested related changes)

**Review by georgiee — APPROVED** [review]
Thank you for the lovely addition and the first step towards our more friendly/warm design. There is one issue with a wrong key `blogpos`, you can use my suggestions.

I also commented on not using my email, and instead use the existing email for applications. The idea to mention unsolicited applications is good though 👍

I also mentioned Gholam to get feedback on his mentions.

This change is good. Let's get the feedback from Gholam before merging but otherwise good. I'm happy to approve this change including the translations.

## Sidenote regarding the design
The design of the icons doesn't work on dark backgrounds. Does the design even cover this case?

here good vs bad. The dark background swallows the icons. 

Screen Shot 2021-11-08 at 09 44 23

Screen Shot 2021-11-08 at 09 45 56

File: src/templates/blog-post.tsx
Comment (hofmann-felix) [reply src/templates/blog-post.tsx]: Thanks for your feedback 🙏
I like the idea with the customizable leadbox for blogposts. I will add a corresponding field to the blogpost template

File: src/components/leadbox/leadbox.stories.tsx
Comment (fischeversenker) [inline src/components/leadbox/leadbox.stories.tsx]: Nice that you added the `fullscreen` layout 👌 
When I tested this in storybook I desperately wanted to change the icon so I had a look at [the documentation](https://storybook.js.org/docs/react/essentials/controls#dealing-with-complex-values) and it turns out we can have arbitrary values in the controls. With some more configuration we would be able to turn the `icon` into a control as well:
```suggestion
import { Astronaut } from '../icons/illustrations/astronaut';
import { SpaceShuttle } from '../icons/illustrations/space-shuttle';

const icons = { Astronaut: (<Astronaut />), SpaceShuttle: (<SpaceShuttle />) };

export default {
  component: Leadbox,
  title: 'Components/Leadbox',
  parameters: {
    layout: 'fullscreen',
  },
  argTypes: {
    icon: {
      options: Object.keys(icons),
      mapping: icons,
    }
  }
} as ComponentMeta<typeof Leadbox>;
```

File: src/components/leadbox/leadbox.tsx
Comment (fischeversenker) [inline src/components/leadbox/leadbox.tsx]: It looks like the grid has some padding that leads to a horizontal scroll bar on small screens. I wasn't able to reproduce this in the actual website, only storybook, but it might still be worth investigation. Adding the `nested` property to the grid solved it for me.

File: src/components/typography/typography.tsx
Comment (fischeversenker) [inline src/components/typography/typography.tsx:191]: I was wondering if we could reuse existing typography settings here instead of introducing new ones. These last two ones look very similar to `ImageCardTitle` and `ImageCardSubtitle`. Maybe we can combine some of the existing ones so we don't have to deal with so many different typography options? I remember Eric had some overview over all the used type faces, maybe we can ask him to receive a definitive list of everything that is used on the website and clean this file up.

**Review by fischeversenker — APPROVED** [review]
Nice new component, thanks :pray:

I left a few comments. None of them block this PR so I'm approving as well. You might still want to have a look at them as long as we wait for Gholams approval for using the mail address 😂

File: src/locales/de/translations.json
Comment (georgiee) [reply src/locales/de/translations.json]: @hofmann-felix you should resolve every positive comment where an actual todo is missing to clear the feedback queue. That way it gets useful again for you 🤓

**Comment by georgiee** [comment]
@hofmann-felix 
One advice, as PRs tends to get out of control and too big if you try to push in every change & idea that arose from the feedback. 

if you are able to collect the remaining issues you want to tackle and work on in a list and merge this PR now (hey, you have two approvals 🥳) you should really do this. It's much easier for your brain. For once, you are starting "fresh" and you don't have two very long feedback streams you have to communicate back the changes. In the worst case your changes will cause other issues that will prevent this PR from being merged as we have to block them until it's fixed. You might see the vicious circle here 😳 Instead you should create a fresh PR an address every change in the PR description and address us in there again.

File: src/locales/de/translations.json
Comment (georgiee) [reply src/locales/de/translations.json]: I spoke to Gholam and we should not expose personal emails yet. Let's stick to `info@satellytes` for anything related to gholam.

**Pull request merged**
